src/processes/helper/sendTopicAutoMessage.js

```diff
@@ -0,0 +1,61 @@
+const momentTz = require('moment-timezone');
+const {v4: uuidv4} = require('uuid');
+const {sample} = require('lodash');
+const {followTopicQueue} = require('../../config/redis');
+const {CommunityMessageFormat, LogError} = require('../../databases/models');
+
+const followTopicServiceQueue = async (userId, topicId, communityMessageFormatId, delay) => {
+  const data = {
+    userId,
+    topicId,
+    communityMessageFormatId
+  };
+  const currentTime = momentTz().tz('America/Los_Angeles');
+  const randomTime = sample([6, 7, 8]);
+  const additionalDays = delay;
+
+  const requiredTime = momentTz()
+    .tz('America/Los_Angeles')
+    .set({hour: randomTime})
+    .add(additionalDays, 'days');
+
+  const diffTime = requiredTime.diff(currentTime, 'milliseconds');
+
+  const options = {
+    jobId: uuidv4(),
+    removeOnComplete: true,
+    delay: diffTime
+  };
+  try {
+    await followTopicQueue.add(data, options);
+  } catch (e) {
+    console.log('error', e);
+  }
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `followTopicServiceQueue` function is missing error handling. If an error occurs while adding a job to the queue, it's only logged to the console and not handled further. This could lead to silent failures that are hard to debug. Consider throwing the error or handling it in a way that's appropriate for your application.

```diff
  try {
    await followTopicQueue.add(data, options);
  } catch (e) {
-   console.log('error', e);
+   throw e;
  }
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->